code/modules/wiremod/components/list/list_pick.dm

```diff
@@ -0,0 +1,77 @@
+/**
+ * # List pick component
+ *
+ * Allows user to select 1 entry from a list
+ */
+
+/obj/item/circuit_component/list_pick
+	display_name = "List Pick"
+	desc = "A component that lets a user pick 1 element from a list. Returns the selected element."
+	category = "List"
+
+	var/datum/port/input/option/list_options
+
+	var/datum/port/input/input_list
+	var/datum/port/input/user
+	var/datum/port/input/input_name
+
+
+	var/datum/port/output/output
+	var/datum/port/output/succes
```

Comment:
please add comments above your vars for [Autodoc](https://github.com/tgstation/tgstation/blob/master/.github/guides/AUTODOC.md)
same applies for assoc list pick